# Q1. What is the Probability density function?
The probability density function (PDF) is a function that describes the relative likelihood of a random variable taking on a certain value. In other words, it provides a mathematical description of the probability distribution of a continuous random variable.

The PDF is defined as the derivative of the cumulative distribution function (CDF), which is a function that gives the probability that a random variable is less than or equal to a certain value. The PDF is represented by a curve, and the area under the curve between two points represents the probability that the random variable will take on a value within that range.

The properties of the PDF depend on the type of probability distribution being considered. For example, the PDF of a normal distribution is a bell-shaped curve, while the PDF of a uniform distribution is a flat line. The PDF is useful in many applications, including statistical inference, risk analysis, and machine learning.

# Q2. What are the types of Probability distribution?
There are many different types of probability distributions, but some of the most commonly used distributions in statistics and probability theory include:

1. Normal distribution: Also known as the Gaussian distribution, this is a continuous probability distribution that is symmetric and bell-shaped. It is often used to model real-world phenomena, such as the heights or weights of people, and many statistical tests assume that data follows a normal distribution.

2. Binomial distribution: This is a discrete probability distribution that describes the number of successes in a fixed number of independent trials, each with the same probability of success. For example, it can be used to model the number of heads in a series of coin flips.

3. Poisson distribution: This is a discrete probability distribution that describes the number of occurrences of a rare event in a fixed amount of time or space. It is often used to model the number of customers arriving at a store or the number of defects in a manufacturing process.

4. Exponential distribution: This is a continuous probability distribution that describes the time between occurrences of a rare event in a Poisson process. It is often used to model waiting times or survival times.

5. Uniform distribution: This is a continuous probability distribution where all values between two endpoints are equally likely. It is often used to model random numbers or events that have no specific bias or preference.

There are many other probability distributions that are used in different fields, such as the Student's t-distribution, the chi-squared distribution, and the Weibull distribution, among others.

# Q3. Write a Python function to calculate the probability density function of a normal distribution with given mean and standard deviation at a given point.


Here's an example Python function that calculates the probability density function of a normal distribution at a given point, given its mean and standard deviation using the SciPy library:

In [1]:
from scipy.stats import norm

def pdf_normal_distribution(x, mean, std_dev):
    # Calculate the probability density function of a normal distribution at a given point
    # with a given mean and standard deviation
    pdf = norm.pdf(x, loc=mean, scale=std_dev)
    return pdf



To use this function, you would provide a value for x, which represents the point at which you want to evaluate the PDF, as well as the mean and std_dev parameters of the normal distribution. For example:

In [2]:
pdf_normal_distribution(2.5, 0, 1)


0.01752830049356854

This would return the PDF of a normal distribution with mean 0 and standard deviation 1, evaluated at x=2.5.

# Q4. What are the properties of Binomial distribution? Give two examples of events where binomial distribution can be applied.
The properties of a binomial distribution are:

1. The experiment consists of a fixed number of trials, denoted by n.
2. Each trial has only two possible outcomes, which are usually called "success" and "failure".
3. The probability of success is constant for each trial and denoted by p.
4. The trials are independent of each other.

Two examples of events where a binomial distribution can be applied are:

1. Flipping a coin: Suppose you flip a fair coin 10 times and want to know the probability of getting exactly 5 heads. This can be modeled using a binomial distribution, where n=10, p=0.5 (since the coin is fair and the probability of heads is 0.5), and the number of successes (k) is 5.
2. Product quality control: Suppose a manufacturer wants to test the quality of a product by inspecting a sample of 50 items and counting the number of defective items. If the defect rate is known to be 10%, then the probability of getting a certain number of defective items can be modeled using a binomial distribution, where n=50, p=0.1, and the number of successes (k) is the number of defective items in the sample.




# Q6. Write a Python function to calculate the cumulative distribution function of a Poisson distribution with given mean at a given point.
Sure, here's a Python function that calculates the cumulative distribution function (CDF) of a Poisson distribution with a given mean mu at a given point k:

In [4]:
import math

def poisson_cdf(mu, k):
    """Calculate the cumulative distribution function (CDF) of a Poisson distribution with mean mu at point k."""
    cdf = 0
    for i in range(k+1):
        cdf += math.exp(-mu) * mu**i / math.factorial(i)
    return cdf


This function uses the formula for the Poisson distribution CDF, which is:

CDF(k) = sum(i=0 to k) exp(-mu) * mu^i / i!

where mu is the mean and k is the point at which the CDF is calculated.

The math module is used to calculate the exponential and factorial functions needed for the formula.

Here's an example of how to use the function:

In [8]:
poisson_cdf(3, 2)


0.42319008112684353

This means that the probability of observing 2 or fewer events in a Poisson process with a mean of 3 is approximately 0.423.

# Q7. How Binomial distribution different from Poisson distribution?
Binomial distribution and Poisson distribution are two commonly used probability distributions, but they have different characteristics and are used in different situations.

The main differences between the two distributions are:

1. The Binomial distribution is used when there are a fixed number of trials, each trial is independent, and there are only two possible outcomes (success or failure). The Poisson distribution is used when the number of events in a fixed interval of time or space follows a Poisson process, where the events occur independently and at a constant rate.

2. In the Binomial distribution, the probability of success (p) is constant for each trial. In the Poisson distribution, the mean (mu) is constant, but the probability of observing a certain number of events in a given interval depends on the length of the interval.

3. The Binomial distribution is discrete, meaning that it is defined only for integer values of the number of successes. The Poisson distribution is also discrete, but it can be approximated by a continuous distribution (the normal distribution) for large values of mu.

4. The Binomial distribution has two parameters: the number of trials (n) and the probability of success (p). The Poisson distribution has only one parameter: the mean (mu).

5. The Binomial distribution is symmetric for p=0.5, meaning that the probability of getting k successes is the same as the probability of getting n-k failures. The Poisson distribution is skewed to the right, meaning that it has a longer tail on the right-hand side.

In summary, the Binomial distribution is used for counting the number of successes in a fixed number of independent trials, while the Poisson distribution is used for counting the number of events in a fixed interval of time or space in a Poisson process.





# Q8. Generate a random sample of size 1000 from a Poisson distribution with mean 5 and calculate the sample mean and variance.

In [9]:
import numpy as np

# Set seed for reproducibility
np.random.seed(123)

# Generate random sample of size 1000 from Poisson distribution with mean 5
sample = np.random.poisson(lam=5, size=1000)

# Calculate sample mean and variance
sample_mean = np.mean(sample)
sample_variance = np.var(sample)

print("Sample mean:", sample_mean)
print("Sample variance:", sample_variance)


Sample mean: 4.919
Sample variance: 5.030439


# Q9. How mean and variance are related in Binomial distribution and Poisson distribution?
In both the Binomial and Poisson distributions, the mean and variance are related in a similar way:

For a Binomial distribution with parameters n (number of trials) and p (probability of success), the mean and variance are given by:

mean = n * p
variance = n * p * (1 - p)
For a Poisson distribution with parameter mu (mean), the mean and variance are both equal to mu:


mean = mu
variance = mu
So, in both distributions, the variance is a function of the mean. For the Binomial distribution, the variance depends on both n and p, while for the Poisson distribution, the variance is solely determined by the mean mu.

In other words, as the mean of the Binomial or Poisson distribution increases, so does its variance. However, for the Poisson distribution, the mean and variance are always equal, while for the Binomial distribution, the variance depends on the value of p and the number of trials n.





# Q10. In normal distribution with respect to mean position, where does the least frequent data appear?
In a normal distribution, the least frequent data appears in the tails of the distribution, which are the areas furthest away from the mean. Specifically, the least frequent data appears at the extremes of the distribution, beyond a certain number of standard deviations from the mean.

For a normal distribution, about 68% of the data falls within one standard deviation of the mean, about 95% falls within two standard deviations of the mean, and about 99.7% falls within three standard deviations of the mean. This means that the data in the tails, beyond three standard deviations from the mean, is very infrequent, with a probability of less than 0.3%.

For example, if a normal distribution has a mean of 50 and a standard deviation of 10, the least frequent data would be found beyond 3 standard deviations from the mean, which would be beyond a value of 80 or below a value of 20.



